src/ZkFold/Base/Protocol/ARK/Plonk.hs

```diff
@@ -38,14 +39,14 @@ type G2 = Point BLS12_381_G2
     NOTE: we need to parametrize the type of transcripts because we use BuiltinByteString on-chain and ByteString off-chain.
     Additionally, we don't want this library to depend on Cardano libraries.
 -}
-data Plonk (d :: Natural) t = Plonk F F F (Map Natural F) (ArithmeticCircuit F) F
+data Plonk (d :: Natural) t = Plonk F F F Natural (V.Vector Natural) (ArithmeticCircuit F) F
```

Comment:
Can we use `ZkFold.Base.Data.Vector.Vector` here? It has the size type parameter in it.